src/Mod/Material/App/Library.cpp

```diff
@@ -0,0 +1,143 @@
+/***************************************************************************
+ *   Copyright (c) 2023 David Carter <dcarter@david.carter.ca>             *
+ *                                                                         *
+ *   This file is part of FreeCAD.                                         *
+ *                                                                         *
+ *   FreeCAD is free software: you can redistribute it and/or modify it    *
+ *   under the terms of the GNU Lesser General Public License as           *
+ *   published by the Free Software Foundation, either version 2.1 of the  *
+ *   License, or (at your option) any later version.                       *
+ *                                                                         *
+ *   FreeCAD is distributed in the hope that it will be useful, but        *
+ *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU      *
+ *   Lesser General Public License for more details.                       *
+ *                                                                         *
+ *   You should have received a copy of the GNU Lesser General Public      *
+ *   License along with FreeCAD. If not, see                               *
+ *   <https://www.gnu.org/licenses/>.                                      *
+ *                                                                         *
+ **************************************************************************/
+
+#include "PreCompiled.h"
+#ifndef _PreComp_
+#include <string>
+#endif
+
+#include <App/Application.h>
+
+#include "Exceptions.h"
+#include "Library.h"
+
+
+using namespace Materials;
+
+TYPESYSTEM_SOURCE(Materials::Library, Base::BaseClass)
+
+Library::Library(const QString& libraryName, const QString& icon, bool readOnly)
+    : _name(libraryName)
+    , _iconPath(icon)
+    , _readOnly(readOnly)
+{}
+
+Library::Library(const QString& libraryName, const QString& dir, const QString& icon, bool readOnly)
+    : _name(libraryName)
+    , _iconPath(icon)
+    , _readOnly(readOnly)
+    , _directory(QDir::cleanPath(dir))
```

Comment:
Compiler warning: wrong order
```suggestion
    : _name(libraryName)
    , _directory(QDir::cleanPath(dir))
    , _iconPath(icon)
    , _readOnly(readOnly)
```